cmd/harmony/main.go

```diff
@@ -101,8 +100,11 @@ var (
 	isExplorer = flag.Bool("is_explorer", false, "true means this node is a node to serve explorer")
 	// networkType indicates the type of the network
 	networkType = flag.String("network_type", "mainnet", "type of the network: mainnet, testnet, devnet...")
+	// blockPeriod indicates the how long the leader waits to propose a new block.
+	blockPeriod = flag.Int("block_period", 10, "how long in second the leader waits to propose a new block.")
```

Comment:
This still won't gaurantee exact 10s block interval. It only adds 10s delay after the consensus is reached. If consensus take longer, the block interval will be longer. The best solution is to have a timer for 10s starting from consensus start and propose the next block when the 10s timeout is reached and consensus was reached before 10s. If consensus takes more than 10s, then propose next block immediately thereafter.

Reply:
agree, that's the best one.  @mikedoan , can you add another change to support that?